particle-iot/device-os
# Implement Mesh.selectAntenna()

### Problem

Even though the antenna is shared between mesh and BLE on gen3 devices, for parity with `BLE` there should be a method in the `Mesh` class that would allow setting the antenna used for mesh radio. This setting should be stored persistently, as otherwise an endpoint mesh device won't be able to connect to the cloud in safe mode.

### Solution

Implement `Mesh.selectAntenna()` and use DCT to store the antenna setting persistently.

This PR includes commits from https://github.com/particle-iot/device-os/pull/1693 and fixes https://github.com/particle-iot/device-os/issues/1861.

### Steps to Test

A simple test would be to select the external antenna on a Xenon device and verify that the device is able to connect to the cloud in safe mode. However, in my tests a Xenon is consistently able to connect to its mesh network even without an antenna attached. Any suggestions are welcomed :smile:

### Example App

```cpp
STARTUP(Mesh.selectAntenna(MeshAntennaType::EXTERNAL))

void setup() {
}

void loop() {
}
```

### References

- [ch38847]
- Closes #1861 
- Closes #1693
